torchrl/envs/common.py

```diff
@@ -1585,20 +1569,14 @@ def policy(td):
             tensordict = self.step(tensordict)
             tensordicts.append(tensordict.clone(False))
 
-            any_done = False
-            _reset_map = {}
-            for done_key in self.done_keys:
-                done = tensordict.get(("next", done_key))
-                truncated = tensordict.get(
-                    ("next", _replace_last(done_key, "truncated")),
-                    default=torch.zeros((), device=done.device, dtype=torch.bool),
-                )
-                done = done | truncated
-                any_sub_done = done.any().item()
-                if any_sub_done and not break_when_any_done:
-                    # Add this done to the map, we will need it to reset
-                    _reset_map.update({_replace_last(done_key, "_reset"): done})
-                any_done += any_sub_done
```

Comment:
this was mostly for efficiency reasons. the reason behind it is that we write a _reset only if there are any dones.